src/autopas/LogicHandler.h

```diff
@@ -601,21 +623,53 @@ class LogicHandler {
   };
 
   /**
-   * Triggers the core steps of the pairwise iteration:
+   * Triggers the core steps of computing the particle interactions:
    *    - functor init- / end traversal
    *    - rebuilding of neighbor lists
-   *    - container.iteratePairwise()
+   *    - container.computeInteractions()
    *    - remainder traversal
    *    - time and energy measurements.
    *
-   * @tparam PairwiseFunctor
+   * @tparam Functor
    * @param functor
    * @param traversal
    * @return Struct containing time and energy measurements. If no energy measurements were possible the respective
    * fields are filled with NaN.
    */
-  template <class PairwiseFunctor>
-  IterationMeasurements iteratePairwise(PairwiseFunctor &functor, TraversalInterface &traversal);
+  template <class Functor>
+  IterationMeasurements computeInteractions(Functor &functor, TraversalInterface &traversal);
+
+  /**
+   * Rebuild the current container's 2-body or 3-body neighbor lists.
+   *
+   * @tparam interactionType
+   * @param traversal
+   * @return
+   */
+  template <InteractionTypeOption::Value interactionType>
+  void rebuildNeighborLists(TraversalInterface &traversal);
```

Comment:
What is the advantage of making `interactionType` a template parameter instead of a function argument here and in the following functions?

Reply:
I guess none. Usually I would prefer it over dynamic casting, but since I do that anyway, the template is not needed. [f0604fd](https://github.com/AutoPas/AutoPas/pull/751/commits/f0604fdf93b487ebeb487e6c5ea0da8663bc92b3)